client/state/user-profile-links/reducer.js

```diff
@@ -0,0 +1,49 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { reject } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer } from 'state/utils';
+import {
+	USER_PROFILE_LINKS_ADD_DUPLICATE,
+	USER_PROFILE_LINKS_ADD_FAILURE,
+	USER_PROFILE_LINKS_ADD_MALFORMED,
+	USER_PROFILE_LINKS_ADD_SUCCESS,
+	USER_PROFILE_LINKS_DELETE_FAILURE,
+	USER_PROFILE_LINKS_DELETE_SUCCESS,
+	USER_PROFILE_LINKS_RECEIVE,
+	USER_PROFILE_LINKS_RESET_ERRORS,
+} from 'state/action-types';
+
+export const items = createReducer( [], {
+	[ USER_PROFILE_LINKS_RECEIVE ]: ( state, { profileLinks } ) => profileLinks,
+	[ USER_PROFILE_LINKS_DELETE_SUCCESS ]: ( state, { linkSlug } ) =>
+		reject( state, { link_slug: linkSlug } ),
+} );
+
+export const errors = createReducer(
+	{},
```

Comment:
Depending on how you use the errors you may not need this reducer. Notices can be shown from the data layer handlers or from the dedicated [notice middleware](https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/middleware.js).

Reply:
Yes, good call. Actually, that was the initial plan, but then I saw that we don't use global notices for user profile links - we rather display them inline:

![](https://cldup.com/PqA4qnJpWO.png)

![](https://cldup.com/XmY8RRmAL8.png)

So we need those errors to be stored in the Redux state tree - I don't really want to introduce any behavioral/visual changes with this refactor.

Reply:
👍